neard/src/runtime.rs

```diff
@@ -770,6 +773,47 @@ impl RuntimeAdapter for NightshadeRuntime {
         }
     }
 
+    #[cfg(feature = "protocol_feature_block_header_v3")]
+    fn verify_approvals_and_threshold_orphan(
+        &self,
+        epoch_id: &EpochId,
+        doomslug_threshold_mode: DoomslugThresholdMode,
+        prev_block_hash: &CryptoHash,
+        prev_block_height: BlockHeight,
+        block_height: BlockHeight,
+        approvals: &[Option<Signature>],
+    ) -> Result<(), Error> {
+        let mut epoch_manager = self.epoch_manager.as_ref().write().expect(POISONED_LOCK_ERR);
+        let info =
```

Comment:
let's move this two lines in two a block so that the lock is dropped immediately after we get the necessary data.